sirepo/package_data/static/js/sirepo-plotting.js

```
@@ -3052,992 +3071,110 @@ SIREPO.app.directive('particle', function(plotting, layoutService, utilities) {
-                showActors(lineActors, $scope.showAbsorbed);
-                showActors(impactSphereActors, $scope.showAbsorbed && $scope.showImpact);
-            };
-            $scope.toggleImpact = function() {
-                $scope.showImpact = ! $scope.showImpact;
-                showActors(impactSphereActors, $scope.showAbsorbed && $scope.showImpact);
-            };
-            $scope.toggleReflected = function() {
-                $scope.showReflected = ! $scope.showReflected;
-                showActors(reflectedLineActors, $scope.showReflected);
-            };
-            $scope.toggleConductors = function() {
-                $scope.showConductors = ! $scope.showConductors;
-                showActors(conductorActors, $scope.showConductors, 0.80);
-            };
-            function showActors(actorArray, doShow, visibleOpacity, hiddenOpacity) {
-                for(var aIndex = 0; aIndex < actorArray.length; ++aIndex) {
-                    actorArray[aIndex].getProperty().setOpacity(doShow ? visibleOpacity || 1.0 : hiddenOpacity || 0.0);
-                }
-                renderWindow.render();
-            }
-
-            function select(selector) {
-                var e = d3.select($scope.element);
-                return selector ? e.select(selector) : e;
+    // Returns edges that fit entirely within the given bounds (if any)
+    this.edgesInsideBounds = function(edges, bounds) {
+        var clippedEnds = [];
+        for(var edge in edges) {
+            var p = edges[edge];
+            if(! self.isPointWithinBounds(p, bounds)) {
+                continue;
             }
+            clippedEnds.push(p);
+        }
+        return clippedEnds;
+    };
 
+    // Sort (with optional reversal) the point array by the values in the given dimension;
+    // Array is cloned first so the original is unchanged
+    this.sortInDimension = function (pArr, dim, reverse) {
+        if(!pArr || !pArr.length || dim >= pArr[0].length ) {
```

seems like this would be an error. I don't understand why null is a reasonable result of a sort. Indeed, if a null is ever returned, the code will bomb later, which is not fail fast. Less code here would make it more reliable, because an error would throw an exception.